### PR TITLE
Command: remove slashes from commands

### PR DESCRIPTION
**Pull request by abeatrix**
CLOSE https://github.com/sourcegraph/cody/issues/1730

Remove slashes from command names and implemented design from https://www.figma.com/file/jEsnzgsf0hNuJbqB9pKaGm/VS-Code---Cody---Commands-up-front-and-ChatGPT%2B%2B?type=design&node-id=1550-62492&mode=design&t=qdjbLWsl9qHq578p-4

Main Commands Menu

image

Custom Commands Menu

image

image

image

## Test plan

Updated all current tests.

https://github.com/sourcegraph/cody/assets/68532117/581ef1e9-c0bc-461d-83ae-ffc03098d2fc

### Before 

image

image

### CODE REVIEW

File: vscode/src/commands/CommandsController.ts [hunk 1]
```diff
@@ -34,7 +35,7 @@ class CommandsController implements vscode.Disposable {
         const commandSplit = text?.trim().split(' ')
         // The unique key for the command. e.g. /test
         const commandKey = commandSplit?.shift() || text
-        const command = this.provider?.get(commandKey)
+        const command = this.provider?.get(fromSlashCommand(commandKey))
```
Comment (valerybugakov) [inline vscode/src/commands/CommandsController.ts:38]: Is this required for backward compat?

File: vscode/src/commands/menus/index.ts [hunk 2]
```diff
@@ -25,7 +26,7 @@ export async function showCommandMenu(
         if (type === 'default') {
             items.push(CommandMenuSeperator.commands)
             for (const [_name, _command] of vscodeDefaultCommands) {
-                const label = _command.slashCommand
+                const label = fromSlashCommand(_command.slashCommand)
```
Comment (valerybugakov) [inline vscode/src/commands/menus/index.ts:29]: Does it make sense to deprecate `CodyCommand.slashCommand` and introduce `CodyCommand.command` or smth like this to reflect the updated behavior better?

Comment (abeatrix) [reply vscode/src/commands/menus/index.ts]: good point, i've deprecated slashCommand and replace it with key instead

**Comment by toolmantim** [comment]
Design wise we want to have it match the commands treeview.

Could we update both the treeview and the command menu to match this?
https://www.figma.com/file/jEsnzgsf0hNuJbqB9pKaGm/VS-Code---Cody---Commands-up-front-and-ChatGPT%2B%2B?type=design&node-id=1550-62492&mode=design&t=qdjbLWsl9qHq578p-4

**Comment by abeatrix** [comment]
@toolmantim updated and ready for review

**Review by toolmantim — COMMENTED** [review]
This is great!

A few small changes if we could:

* "New Chat" from the quickpick should act the same as "New Chat" in the review (open a new chat tab)
* Could we add the keyboard shortcuts for "New Chat" and "Edit Code" to the quickpick item descriptions?
* If there are no custom commands, can we make "Custom Commands" quickpick item & treeview item go directly to the "Configure Custom Commands…" and skip the empty-ish middle quickpick?

One bug:

* In the quickpick "Custom Commands" > "Configure Custom Commands…" both the "New Custom Command..." and "Open Custom Commands Documentation" items don't seem to work

**Comment by abeatrix** [comment]
@toolmantim updated based on your suggestions and fixed the bug:
- [x] "New Chat" from the quickpick should act the same as "New Chat" in the review (open a new chat tab)
- [x] Could we add the keyboard shortcuts for "New Chat" and "Edit Code" to the quickpick item descriptions?
  - [x] Also added a keyboard shortcuts for "Custom Commands" as shown in your figma design
- [x] If there are no custom commands, can we make "Custom Commands" quickpick item & treeview item go directly to the "Configure Custom Commands…" and skip the empty-ish middle quickpick?
- [x] Fixed bug: In the quickpick "Custom Commands" > "Configure Custom Commands…" both the "New Custom Command..." and "Open Custom Commands Documentation" items don't seem to work

**Review by toolmantim — APPROVED** [review]
Nice! Two non-blocking things…

If you start typing "New" you end up seeing duplicates. I'd suggest we always exclude the "New Chat (shortcut)" and "Edit Code (shortcut)" items when you start filtering/typing… because they're always going to be there as "Start a new chat" and "Start a new edit" anyhow.

CleanShot 2024-02-13 at 16 18 16@2x

Also when you filter in custom commands we probably don't want to show these last 3 items, just the filtered custom commands (or nothing if no matches):

CleanShot 2024-02-13 at 16 18 40@2x

**Comment by toolmantim** [comment]
Does this binding actually exist? I can't get it to trigger:

CleanShot 2024-02-13 at 16 24 19@2x

If it doesn't exist, we can just remove that shortcut label for now.

**Comment by abeatrix** [comment]
> Does this binding actually exist? I can't get it to trigger:
> 
> CleanShot 2024-02-13 at 16 24 19@2x
> 
> If it doesn't exist, we can just remove that shortcut label for now.

it does exist! I just fixed this and the bugs you found. Will update the e2e tests tomorrow to cover these cases before I finalize the PR tomorrow, thanks for taking the time to review and test this PR out!

File: vscode/src/commands/index.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import os from 'os'
+import type { CodySidebarTreeItem } from '../services/treeViewItems'
+
+const isMac = os.platform() === 'darwin'
```
Comment (valerybugakov) [inline vscode/src/commands/index.ts:4]: Nit: we have multiple helpers like this? Is it worth reusing the same implementation everywhere? 
Screenshot 2024-02-13 at 15 23 57

File: vscode/src/commands/menus/types.ts [hunk 4]
```diff
@@ -20,8 +20,15 @@ export interface CommandMenuItem extends QuickPickItem {
     type?: CodyCommandType
     // vs code command, e.g. 'cody.commands.open.json'
     command?: string
-    // cody command, e.g. '/ask'
+    /**
+     * cody command, e.g. '/ask'
+     * @deprecated Use 'commandKey' instead.
+     */
     slashCommand?: string
```
Comment (valerybugakov) [inline vscode/src/commands/menus/types.ts:27]: Nice! 🙌

**Review by valerybugakov — APPROVED** [review]
Looks great!

File: vscode/src/commands/index.ts (hunk 3 above)
Comment (abeatrix) [reply vscode/src/commands/index.ts]: Good catch! I've move the one for node to lib/shared!

**Pull request merged**